### ISSUE TITLE
Nodetool decommission doesn't make a decommissioned node be non-voting which makes CL=ALL stop succeeding

### ISSUE THREAD

**Issue by vponomaryov** [scylladb/scylla#10119]
_Installation details_ 
Kernel version: `5.11.0-1028-aws` 
Scylla version (or git commit hash): `5.1.dev-0.20220217.69fcc053b with build-id b8415b1ebbffff2b4183734680f4afab3bfed86d` 
Cluster size: 6 nodes (r5b.2xlarge) 
OS (RHEL/CentOS/Ubuntu/AWS AMI): `ami-041d8500e7cf30167` (aws: eu-east-1)

Scylla running with shards number (live nodes): 
longevity-100gb-4h-master-db-node-490f8141-1 (54.195.177.198 | 10.0.1.17), shards: 8
longevity-100gb-4h-master-db-node-490f8141-3 (3.250.213.212 | 10.0.3.235), shards: 8
longevity-100gb-4h-master-db-node-490f8141-4 (3.248.215.55 | 10.0.2.60), shards: 8
longevity-100gb-4h-master-db-node-490f8141-5 (34.240.24.189 | 10.0.3.86), shards: 8
longevity-100gb-4h-master-db-node-490f8141-6 (54.72.208.32 | 10.0.2.192), shards: 8
longevity-100gb-4h-master-db-node-490f8141-7 (52.210.129.160 | 10.0.2.161), shards: 8

Dead nodes:
longevity-100gb-4h-master-db-node-490f8141-2 (54.171.17.182 | 10.0.1.235), shards: 8, 
	terminated at 2022-02-20 05:31:27.117362
	terminated by 'NodetoolDecommission' nemesis

Test: `longevity_test.LongevityTest.test_custom_time` Test name: `performance_regression_test.PerformanceRegressionTest.test_latency` Test config file(s):
* [longevity-100gb-4h.yaml](https://github.com/scylladb/scylla-cluster-tests/blob/60aaec5e740928c616cecd72b622c3c4214c822e/test-cases/longevity/longevity-100gb-4h.yaml)
* [ebs-gp3-2v-4tb-16k-iops.yaml](https://github.com/scylladb/scylla-cluster-tests/blob/60aaec5e740928c616cecd72b622c3c4214c822e/configurations/ebs/ebs-gp3-2v-4tb-16k-iops.yaml)

**Issue description**
Nemesis 'disrupt_nodetool_decommission' has following steps:
- Runs the 'nodetool decommission' command
- Waits for the start of the 'decommission' node operation
- Waits for the end of the 'decommission' node operation
- Adds new node to a Scylla cluster

So, here is the first 3 steps where `longevity-100gb-4h-master-db-node-490f8141-2` gets decommissioned spending 19 minutes:

```
2022-02-20T05:12:16+00:00 db-node-490f8141-2 !    INFO |  [shard 0] storage_service - DECOMMISSIONING: starts
2022-02-20T05:12:16+00:00 db-node-490f8141-2 !    INFO |  [shard 0] storage_service - decommission[281195e3-7c7d-458a-a68c-0d73f2a1ca33]: Started decommission operation,
                                                                        removing node=10.0.1.235, sync_nodes=[10.0.3.86, 10.0.2.192, 10.0.3.235, 10.0.2.60, 10.0.1.17, 10.0.1.235], ignore_nodes=[]
2022-02-20T05:12:16+00:00 db-node-490f8141-2 !    INFO |  [shard 0] storage_service - decommission[281195e3-7c7d-458a-a68c-0d73f2a1ca33]: Added node=10.0.1.235 as leaving node, coordinator=10.0.1.235
2022-02-20T05:12:16+00:00 db-node-490f8141-2 !    INFO |  [shard 0] storage_service - decommission[281195e3-7c7d-458a-a68c-0d73f2a1ca33]: Started heartbeat_updater
2022-02-20T05:12:16+00:00 db-node-490f8141-2 !    INFO |  [shard 0] storage_service - DECOMMISSIONING: unbootstrap starts
2022-02-20T05:12:16+00:00 db-node-490f8141-2 !    INFO |  [shard 0] storage_service - enable_repair_based_node_ops=true, allowed_repair_based_node_ops={replace}
2022-02-20T05:12:16+00:00 db-node-490f8141-2 !    INFO |  [shard 0] storage_service - LEAVING: replaying batch log and streaming data to other nodes
2022-02-20T05:12:16+00:00 db-node-490f8141-2 !    INFO |  [shard 0] range_streamer - Unbootstrap starts, nr_ranges_remaining=2786
2022-02-20T05:12:16+00:00 db-node-490f8141-2 !    INFO |  [shard 0] range_streamer - Unbootstrap with [10.0.2.192, 10.0.3.235, 10.0.3.86, 10.0.1.17, 10.0.2.60] for keyspace=system_auth started, nodes_to_st      ream=5
2022-02-20T05:12:16+00:00 db-node-490f8141-2 !    INFO |  [shard 0] range_streamer - Unbootstrap with 10.0.2.192 for keyspace=system_auth, streaming [0, 13) out of 134 ranges
2022-02-20T05:12:16+00:00 db-node-490f8141-2 !    INFO |  [shard 0] stream_session - [Stream #ac0188c0-920b-11ec-b971-e2b932f50c99] Executing streaming plan for Unbootstrap-system_auth-index-0 with peers={      10.0.2.192}, master
2022-02-20T05:12:16+00:00 db-node-490f8141-2 !    INFO |  [shard 0] range_streamer - Unbootstrap with 10.0.3.235 for keyspace=system_auth, streaming [0, 13) out of 135 ranges
2022-02-20T05:12:16+00:00 db-node-490f8141-2 !    INFO |  [shard 0] stream_session - [Stream #ac0188c1-920b-11ec-b971-e2b932f50c99] Executing streaming plan for Unbootstrap-system_auth-index-0 with peers={      10.0.3.235}, master
2022-02-20T05:12:16+00:00 db-node-490f8141-2 !    INFO |  [shard 0] range_streamer - Unbootstrap with 10.0.3.86 for keyspace=system_auth, streaming [0, 15) out of 156 ranges
2022-02-20T05:12:16+00:00 db-node-490f8141-2 !    INFO |  [shard 0] stream_session - [Stream #ac0188c2-920b-11ec-b971-e2b932f50c99] Executing streaming plan for Unbootstrap-system_auth-index-0 with peers={      10.0.3.86}, master
2022-02-20T05:12:16+00:00 db-node-490f8141-2 !    INFO |  [shard 0] range_streamer - Unbootstrap with 10.0.1.17 for keyspace=system_auth, streaming [0, 13) out of 137 ranges
2022-02-20T05:12:16+00:00 db-node-490f8141-2 !    INFO |  [shard 0] stream_session - [Stream #ac0188c3-920b-11ec-b971-e2b932f50c99] Executing streaming plan for Unbootstrap-system_auth-index-0 with peers={      10.0.1.17}, master
2022-02-20T05:12:16+00:00 db-node-490f8141-2 !    INFO |  [shard 0] range_streamer - Unbootstrap with 10.0.2.60 for keyspace=system_auth, streaming [0, 19) out of 195 ranges
2022-02-20T05:12:16+00:00 db-node-490f8141-2 !    INFO |  [shard 0] stream_session - [Stream #ac0188c4-920b-11ec-b971-e2b932f50c99] Executing streaming plan for Unbootstrap-system_auth-index-0 with peers={      10.0.2.60}, master
2022-02-20T05:12:16+00:00 db-node-490f8141-2 !    INFO |  [shard 0] stream_session - [Stream #ac0188c3-920b-11ec-b971-e2b932f50c99] Skip sending ks=system_auth, cf=role_members, reader contains no data, wi      th new rpc streaming
...                                                                                                    
2022-02-20T05:31:07+00:00 db-node-490f8141-2 !    INFO |  [shard 0] storage_service - DECOMMISSIONING: unbootstrap done
2022-02-20T05:31:07+00:00 db-node-490f8141-2 !    INFO |  [shard 0] storage_service - decommission[281195e3-7c7d-458a-a68c-0d73f2a1ca33]: Marked ops done from coordinator=10.0.1.235
2022-02-20T05:31:07+00:00 db-node-490f8141-2 !    INFO |  [shard 0] storage_service - decommission[281195e3-7c7d-458a-a68c-0d73f2a1ca33]: Finished decommission operation, 
                                                                        removing node=10.0.1.235, sync_nodes=[10.0.3.86, 10.0.2.192, 10.0.3.235, 10.0.2.60, 10.0.1.17, 10.0.1.235], ignore_nodes=[]
2022-02-20T05:31:07+00:00 db-node-490f8141-2 !    INFO |  [shard 0] storage_service - decommission[281195e3-7c7d-458a-a68c-0d73f2a1ca33]: Stopped heartbeat_updater
2022-02-20T05:31:07+00:00 db-node-490f8141-2 !    INFO |  [shard 0] storage_service - DECOMMISSIONING: leaving Raft group 0
2022-02-20T05:31:07+00:00 db-node-490f8141-2 !    INFO |  [shard 0] storage_service - DECOMMISSIONING: left Raft group 0
2022-02-20T05:31:07+00:00 db-node-490f8141-2 !    INFO |  [shard 0] storage_service - Stop transport: starts
2022-02-20T05:31:07+00:00 db-node-490f8141-2 !    INFO |  [shard 0] storage_service - Shutting down native transport server
2022-02-20T05:31:07+00:00 db-node-490f8141-2 !    INFO |  [shard 0] cql_server_controller - CQL server stopped
2022-02-20T05:31:07+00:00 db-node-490f8141-2 !    INFO |  [shard 0] storage_service - Shutting down native transport server was successful
2022-02-20T05:31:07+00:00 db-node-490f8141-2 !    INFO |  [shard 0] storage_service - Shutting down rpc server
2022-02-20T05:31:07+00:00 db-node-490f8141-2 !    INFO |  [shard 0] storage_service - Shutting down rpc server was successful
2022-02-20T05:31:07+00:00 db-node-490f8141-2 !    INFO |  [shard 0] storage_service - Shutting down alternator server
2022-02-20T05:31:07+00:00 db-node-490f8141-2 !    INFO |  [shard 0] storage_service - Shutting down alternator server was successful
2022-02-20T05:31:07+00:00 db-node-490f8141-2 !    INFO |  [shard 0] storage_service - Shutting down redis server
2022-02-20T05:31:07+00:00 db-node-490f8141-2 !    INFO |  [shard 0] storage_service - Shutting down redis server was successful
2022-02-20T05:31:07+00:00 db-node-490f8141-2 !    INFO |  [shard 0] storage_service - Stop transport: shutdown rpc and cql server done
2022-02-20T05:31:07+00:00 db-node-490f8141-2 !    INFO |  [shard 0] gossip - My status = LEFT          
2022-02-20T05:31:07+00:00 db-node-490f8141-2 ! WARNING |  [shard 0] gossip - No local state or state is in silent shutdown, not announcing shutdown
2022-02-20T05:31:07+00:00 db-node-490f8141-2 !    INFO |  [shard 0] gossip - Disable and wait for gossip loop started
2022-02-20T05:31:07+00:00 db-node-490f8141-2 !    INFO |  [shard 0] gossip - failure_detector_loop: Finished main loop
2022-02-20T05:31:07+00:00 db-node-490f8141-2 !    INFO |  [shard 0] gossip - Gossip is now stopped  
2022-02-20T05:31:07+00:00 db-node-490f8141-2 !    INFO |  [shard 0] storage_service - Stop transport: stop_gossiping done
2022-02-20T05:31:07+00:00 db-node-490f8141-2 !    INFO |  [shard 0] messaging_service - Stopping nontls server
2022-02-20T05:31:07+00:00 db-node-490f8141-2 !    INFO |  [shard 0] messaging_service - Stopping tls server
2022-02-20T05:31:07+00:00 db-node-490f8141-2 !    INFO |  [shard 0] messaging_service - Stopping tls server - Done
2022-02-20T05:31:07+00:00 db-node-490f8141-2 !    INFO |  [shard 0] messaging_service - Stopping client for address: 10.0.2.192:0

...                                                                                                    
2022-02-20T05:31:07+00:00 db-node-490f8141-2 !    INFO |  [shard 0] storage_service - messaging_service stopped
2022-02-20T05:31:07+00:00 db-node-490f8141-2 !    INFO |  [shard 0] storage_service - Stop transport: shutdown messaging_service done
2022-02-20T05:31:07+00:00 db-node-490f8141-2 !    INFO |  [shard 0] storage_service - Stop transport: shutdown stream_manager done
2022-02-20T05:31:07+00:00 db-node-490f8141-2 !    INFO |  [shard 0] storage_service - Stop transport: done
2022-02-20T05:31:07+00:00 db-node-490f8141-2 !    INFO |  [shard 0] storage_service - DECOMMISSIONING: stopped transport
2022-02-20T05:31:07+00:00 db-node-490f8141-2 !    INFO |  [shard 0] storage_service - DECOMMISSIONING: stop batchlog_manager done
2022-02-20T05:31:23+00:00 db-node-490f8141-2 !    INFO |  [shard 0] storage_service - DECOMMISSIONING: set_bootstrap_state done
2022-02-20T05:31:23+00:00 db-node-490f8141-2 !    INFO |  [shard 0] storage_service - DECOMMISSIONED:
2022-02-20T05:31:23+00:00 db-node-490f8141-2 !    INFO |  [shard 0] storage_service - DECOMMISSIONING: done
```

Then the new node gets created. After it we see following logs in it:

```
2022-02-20T05:32:09+00:00 db-node-490f8141-7 !    INFO | Starting System Logging Service...
...
2022-02-20T05:36:47+00:00 db-node-490f8141-7 !    INFO |  [shard 0] gossip - Node 10.0.1.235 will be removed from gossip at [2022-02-23 05:30:30]: (expire = 1645594230175499179, now = 1645335407766195375,       diff = 258822 seconds)
...
2022-02-20T05:36:47+00:00 db-node-490f8141-7 ! WARNING | [shard 0] migration_manager - Can't send migration request: node 10.0.1.235 is down.
...
2022-02-20T05:36:49+00:00 db-node-490f8141-7 !    INFO | [shard 0] compaction - [Compact system_schema.columns 19dd21d0-920f-11ec-ab51-60da59c5b37e] Compacted 3 sstables to [/var/lib/scylla/data/system_sch      ema/columns-24101c25a2ae3af787c1b40ee1aca33f/mc-64-big-Data.db:level=0]. 56kB to 25kB (~45% of original) in 26ms = 984kB/s. ~384 total partitions merged to 7.
2022-02-20T05:36:50+00:00 db-node-490f8141-7 !  NOTICE | scyllaadm : TTY=unknown ; PWD=/home/scyllaadm ; USER=root ; COMMAND=/usr/bin/apt-key adv --keyserver keyserver.ubuntu.com --recv-keys 5E08FBD8B5D6EC      9C
2022-02-20T05:36:50+00:00 db-node-490f8141-7 !  NOTICE | scyllaadm : TTY=unknown ; PWD=/home/scyllaadm ; USER=root ; COMMAND=/usr/bin/apt-key adv --keyserver keyserver.ubuntu.com --recv-keys 5E08FBD8B5D6EC      9C
2022-02-20T05:36:50+00:00 db-node-490f8141-7 !    INFO | pam_unix(sudo:session): session opened for user root by (uid=0)
2022-02-20T05:36:50+00:00 db-node-490f8141-7 !    INFO | pam_unix(sudo:session): session opened for user root by (uid=0)
2022-02-20T05:36:50+00:00 db-node-490f8141-7 !    INFO | Started Scylla Housekeeping restart mode.
2022-02-20T05:36:50+00:00 db-node-490f8141-7 !    INFO | Started Scylla Housekeeping restart mode.
...
2022-02-20T05:38:12+00:00 db-node-490f8141-7 !    INFO | Started Scylla Manager Agent.
...
2022-02-20T05:38:23+00:00 db-node-490f8141-7 !    INFO | pam_unix(sudo:session): session closed for user root
2022-02-20T05:38:36+00:00 db-node-490f8141-7 !    INFO |  [shard 0] init - Shutting down storage service notifications      
...                                                                                                 
2022-02-20T05:38:38+00:00 db-node-490f8141-7 !    INFO | [shard 0] init - Shutting down sighup was successful
2022-02-20T05:38:38+00:00 db-node-490f8141-7 !     ERR | [shard 0] init - Startup failed: exceptions::mutation_write_timeout_exception (Operation timed out for system_distributed_everywhere.cdc_generation_descriptions_v2 - received only 5 responses from 6 CL=ALL.)
2022-02-20T05:38:38+00:00 longevity-100gb-4h-master-db-node-490f8141-7 !    INFO |  [shard 0] init - Shutting down sighup was successful
2022-02-20T05:38:38+00:00 longevity-100gb-4h-master-db-node-490f8141-7 !     ERR |  [shard 0] init - Startup failed: exceptions::mutation_write_timeout_exception (Operation timed out for system_distributed_everywhere.cdc_generation      _descriptions_v2 - received only 5 responses from 6 CL=ALL.)
2022-02-20T05:38:38+00:00 longevity-100gb-4h-master-db-node-490f8141-7 !    INFO | [shard 0] rpc - client 10.0.2.60:55216 msg_id 70:  exception "gate closed" in no_wait handler ignored
2022-02-20T05:38:38+00:00 longevity-100gb-4h-master-db-node-490f8141-7 !    INFO | [shard 0] rpc - client 10.0.2.60:55216 msg_id 70:  exception "gate closed" in no_wait handler ignored
2022-02-20T05:38:38+00:00 longevity-100gb-4h-master-db-node-490f8141-7 ! WARNING | [shard 0] gossip - failure_detector_loop: Got error in the loop, live_nodes=[10.0.2.60, 10.0.1.17, 10.0.2.192, 10.0.3.235, 10.0.3.86]: seastar::sleep_aborted (Sleep is aborted)
2022-02-20T05:38:38+00:00 longevity-100gb-4h-master-db-node-490f8141-7 !    INFO | [shard 0] gossip - failure_detector_loop: Finished main loop
2022-02-20T05:38:38+00:00 longevity-100gb-4h-master-db-node-490f8141-7 !    INFO | [shard 0] gossip - Gossip is now stopped
2022-02-20T05:38:38+00:00 longevity-100gb-4h-master-db-node-490f8141-7 !    INFO | [shard 0] init - Shutting down gossiper was successful
2022-02-20T05:38:38+00:00 longevity-100gb-4h-master-db-node-490f8141-7 ! WARNING | [shard 0] gossip - failure_detector_loop: Got error in the loop, live_nodes=[10.0.2.60, 10.0.1.17, 10.0.2.192, 10.0.3.235, 10.0.3.86]: seastar::sleep_aborted (Sleep is aborted)
2022-02-20T05:38:38+00:00 longevity-100gb-4h-master-db-node-490f8141-7 !    INFO | [shard 0] gossip - failure_detector_loop: Finished main loop
2022-02-20T05:38:38+00:00 longevity-100gb-4h-master-db-node-490f8141-7 !    INFO | [shard 0] gossip - Gossip is now stopped
...                                                                                                 
2022-02-20T05:44:28+00:00 db-node-490f8141-7 !    INFO | Unable to negotiate with 212.193.30.137 port 58932: no matching key exchange method found. Their offer: diffie-hellman-group14-sha1,diffie-hellman-group-exchange-sha1,diffie-hellman-group1-sha1 [preauth]
```

As we can see, gossip was stopped and node stopped any activity being idle.

Restore Monitor Stack command: `$ hydra investigate show-monitor 490f8141-7f93-46c8-a1d9-ceffaa3688f0` 
Restore monitor on AWS instance using [Jenkins job](https://jenkins.scylladb.com/view/QA/job/QA-tools/job/hydra-show-monitor/parambuild/?test_id=490f8141-7f93-46c8-a1d9-ceffaa3688f0) 
Show all stored logs command: `$ hydra investigate show-logs 490f8141-7f93-46c8-a1d9-ceffaa3688f0`

Test id: `490f8141-7f93-46c8-a1d9-ceffaa3688f0`

Logs: 
db-cluster - https://cloudius-jenkins-test.s3.amazonaws.com/490f8141-7f93-46c8-a1d9-ceffaa3688f0/20220220_104921/db-cluster-490f8141.tar.gz
loader-set - https://cloudius-jenkins-test.s3.amazonaws.com/490f8141-7f93-46c8-a1d9-ceffaa3688f0/20220220_104921/loader-set-490f8141.tar.gz
monitor-set - https://cloudius-jenkins-test.s3.amazonaws.com/490f8141-7f93-46c8-a1d9-ceffaa3688f0/20220220_104921/monitor-set-490f8141.tar.gz
sct-runner - https://cloudius-jenkins-test.s3.amazonaws.com/490f8141-7f93-46c8-a1d9-ceffaa3688f0/20220220_104921/sct-runner-490f8141.tar.gz

[Jenkins job URL](https://jenkins.scylladb.com/job/scylla-master/job/longevity/job/longevity-100gb-4h-ebs-gp3-test/35/)

**Comment by aleksbykov** [comment]
Issue reproduced.
*Installation details*
Kernel version: `5.11.0-1028-aws`
Scylla version (or git commit hash): `5.1.dev-0.20220225.680195564de7 with build-id ddb50a4cec4995f08be48bff703067d207209192`
Cluster size: 6 nodes (r5b.2xlarge)
Scylla running with shards number (live nodes):
   longevity-100gb-4h-master-db-node-4cc6ea94-2 (34.242.203.240 | 10.0.0.246): 8 shards
   longevity-100gb-4h-master-db-node-4cc6ea94-3 (3.250.34.191 | 10.0.1.123): 8 shards
   longevity-100gb-4h-master-db-node-4cc6ea94-4 (34.243.117.238 | 10.0.3.95): 8 shards
   longevity-100gb-4h-master-db-node-4cc6ea94-5 (34.245.102.155 | 10.0.0.27): 8 shards
   longevity-100gb-4h-master-db-node-4cc6ea94-6 (34.254.244.106 | 10.0.3.103): 8 shards
   longevity-100gb-4h-master-db-node-4cc6ea94-7 (54.246.146.111 | 10.0.2.204): 8 shards
Scylla running with shards number (terminated nodes):
   longevity-100gb-4h-master-db-node-4cc6ea94-1 (52.50.215.208 | 10.0.0.200): 8 shards
OS (RHEL/CentOS/Ubuntu/AWS AMI): `ami-068c5f920fa103552` (aws: eu-west-1)


Test: `longevity-100gb-4h-ebs-gp3-test`
Test name: `longevity_test.LongevityTest.test_custom_time`
Test config file(s):
- [longevity-100gb-4h.yaml]([https://github.com/scylladb/scylla-cluster-tests/blob/8584390cb25ea782b18acef4a6d77099be58b0bc/test-cases/longevity/longevity-100gb-4h.yaml](https://www.google.com/url?q=https://github.com/scylladb/scylla-cluster-tests/blob/8584390cb25ea782b18acef4a6d77099be58b0bc/test-cases/longevity/longevity-100gb-4h.yaml&source=gmail-html&ust=1646143203457000&usg=AOvVaw2Rjzbj3G0vXEJq9_wRWDnl))
- [ebs-gp3-2v-4tb-16k-iops.yaml]([https://github.com/scylladb/scylla-cluster-tests/blob/9894d5c9bddf5edd80da967f290ec6ad02f24c6a/configurations/ebs/ebs-gp3-2v-4tb-16k-iops.yaml](https://www.google.com/url?q=https://github.com/scylladb/scylla-cluster-tests/blob/9894d5c9bddf5edd80da967f290ec6ad02f24c6a/configurations/ebs/ebs-gp3-2v-4tb-16k-iops.yaml&source=gmail-html&ust=1646143203458000&usg=AOvVaw02UCWzMRAVJN_3mEkSX1UM))


**Issue description**

====================================

PUT ISSUE DESCRIPTION HERE


====================================






Restore Monitor Stack command: `$ hydra investigate show-monitor 4cc6ea94-cd00-4a8b-a6a2-edb789b313e3`
Restore monitor on AWS instance using [Jenkins job]([https://jenkins.scylladb.com/view/QA/job/QA-tools/job/hydra-show-monitor/parambuild/?sct_branch=master&test_id=4cc6ea94-cd00-4a8b-a6a2-edb789b313e3](https://www.google.com/url?q=https://jenkins.scylladb.com/view/QA/job/QA-tools/job/hydra-show-monitor/parambuild/?sct_branch%3Dmaster%26test_id%3D4cc6ea94-cd00-4a8b-a6a2-edb789b313e3&source=gmail-html&ust=1646143203458000&usg=AOvVaw17IR71usbLd1ym2d9yF2Nt))
Show all stored logs command: `$ hydra investigate show-logs 4cc6ea94-cd00-4a8b-a6a2-edb789b313e3`

Test id: `4cc6ea94-cd00-4a8b-a6a2-edb789b313e3`


Logs:
grafana - [[https://cloudius-jenkins-test.s3.amazonaws.com/4cc6ea94-cd00-4a8b-a6a2-edb789b313e3/20220227_102441/grafana-screenshot-longevity-100gb-4h-ebs-gp3-test-scylla-per-server-metrics-nemesis-20220227_102655-longevity-100gb-4h-master-monitor-node-4cc6ea94-1.png](https://cloudius-jenkins-test.s3.amazonaws.com/4cc6ea94-cd00-4a8b-a6a2-edb789b313e3/20220227_102441/grafana-screenshot-longevity-100gb-4h-ebs-gp3-test-scylla-per-server-metrics-nemesis-20220227_102655-longevity-100gb-4h-master-monitor-node-4cc6ea94-1.png)](https://www.google.com/url?q=https://cloudius-jenkins-test.s3.amazonaws.com/4cc6ea94-cd00-4a8b-a6a2-edb789b313e3/20220227_102441/grafana-screenshot-longevity-100gb-4h-ebs-gp3-test-scylla-per-server-metrics-nemesis-20220227_102655-longevity-100gb-4h-master-monitor-node-4cc6ea94-1.png%255D(https://cloudius-jenkins-test.s3.amazonaws.com/4cc6ea94-cd00-4a8b-a6a2-edb789b313e3/20220227_102441/grafana-screenshot-longevity-100gb-4h-ebs-gp3-test-scylla-per-server-metrics-nemesis-20220227_102655-longevity-100gb-4h-master-monitor-node-4cc6ea94-1.png)&source=gmail-html&ust=1646143203458000&usg=AOvVaw10eqM7mSPdjC_SIE6coSrI)
grafana - [[https://cloudius-jenkins-test.s3.amazonaws.com/4cc6ea94-cd00-4a8b-a6a2-edb789b313e3/20220227_102441/grafana-screenshot-overview-20220227_102441-longevity-100gb-4h-master-monitor-node-4cc6ea94-1.png](https://cloudius-jenkins-test.s3.amazonaws.com/4cc6ea94-cd00-4a8b-a6a2-edb789b313e3/20220227_102441/grafana-screenshot-overview-20220227_102441-longevity-100gb-4h-master-monitor-node-4cc6ea94-1.png)](https://www.google.com/url?q=https://cloudius-jenkins-test.s3.amazonaws.com/4cc6ea94-cd00-4a8b-a6a2-edb789b313e3/20220227_102441/grafana-screenshot-overview-20220227_102441-longevity-100gb-4h-master-monitor-node-4cc6ea94-1.png%255D(https://cloudius-jenkins-test.s3.amazonaws.com/4cc6ea94-cd00-4a8b-a6a2-edb789b313e3/20220227_102441/grafana-screenshot-overview-20220227_102441-longevity-100gb-4h-master-monitor-node-4cc6ea94-1.png)&source=gmail-html&ust=1646143203458000&usg=AOvVaw3SQIlANoRIr77W_RueJMYU)
db-cluster - [[https://cloudius-jenkins-test.s3.amazonaws.com/4cc6ea94-cd00-4a8b-a6a2-edb789b313e3/20220227_103929/db-cluster-4cc6ea94.tar.gz](https://cloudius-jenkins-test.s3.amazonaws.com/4cc6ea94-cd00-4a8b-a6a2-edb789b313e3/20220227_103929/db-cluster-4cc6ea94.tar.gz)](https://www.google.com/url?q=https://cloudius-jenkins-test.s3.amazonaws.com/4cc6ea94-cd00-4a8b-a6a2-edb789b313e3/20220227_103929/db-cluster-4cc6ea94.tar.gz%255D(https://cloudius-jenkins-test.s3.amazonaws.com/4cc6ea94-cd00-4a8b-a6a2-edb789b313e3/20220227_103929/db-cluster-4cc6ea94.tar.gz)&source=gmail-html&ust=1646143203458000&usg=AOvVaw2VXgIiJhrZcCzOFzYuLdLc)
loader-set - [[https://cloudius-jenkins-test.s3.amazonaws.com/4cc6ea94-cd00-4a8b-a6a2-edb789b313e3/20220227_103929/loader-set-4cc6ea94.tar.gz](https://cloudius-jenkins-test.s3.amazonaws.com/4cc6ea94-cd00-4a8b-a6a2-edb789b313e3/20220227_103929/loader-set-4cc6ea94.tar.gz)](https://www.google.com/url?q=https://cloudius-jenkins-test.s3.amazonaws.com/4cc6ea94-cd00-4a8b-a6a2-edb789b313e3/20220227_103929/loader-set-4cc6ea94.tar.gz%255D(https://cloudius-jenkins-test.s3.amazonaws.com/4cc6ea94-cd00-4a8b-a6a2-edb789b313e3/20220227_103929/loader-set-4cc6ea94.tar.gz)&source=gmail-html&ust=1646143203458000&usg=AOvVaw0F-_fT3Kuq2UFbvio75Tj8)
monitor-set - [[https://cloudius-jenkins-test.s3.amazonaws.com/4cc6ea94-cd00-4a8b-a6a2-edb789b313e3/20220227_103929/monitor-set-4cc6ea94.tar.gz](https://cloudius-jenkins-test.s3.amazonaws.com/4cc6ea94-cd00-4a8b-a6a2-edb789b313e3/20220227_103929/monitor-set-4cc6ea94.tar.gz)](https://www.google.com/url?q=https://cloudius-jenkins-test.s3.amazonaws.com/4cc6ea94-cd00-4a8b-a6a2-edb789b313e3/20220227_103929/monitor-set-4cc6ea94.tar.gz%255D(https://cloudius-jenkins-test.s3.amazonaws.com/4cc6ea94-cd00-4a8b-a6a2-edb789b313e3/20220227_103929/monitor-set-4cc6ea94.tar.gz)&source=gmail-html&ust=1646143203458000&usg=AOvVaw1dkD6JihSto7SN-auPEoDP)
sct - [[https://cloudius-jenkins-test.s3.amazonaws.com/4cc6ea94-cd00-4a8b-a6a2-edb789b313e3/20220227_103929/sct-runner-4cc6ea94.tar.gz](https://cloudius-jenkins-test.s3.amazonaws.com/4cc6ea94-cd00-4a8b-a6a2-edb789b313e3/20220227_103929/sct-runner-4cc6ea94.tar.gz)](https://www.google.com/url?q=https://cloudius-jenkins-test.s3.amazonaws.com/4cc6ea94-cd00-4a8b-a6a2-edb789b313e3/20220227_103929/sct-runner-4cc6ea94.tar.gz%255D(https://cloudius-jenkins-test.s3.amazonaws.com/4cc6ea94-cd00-4a8b-a6a2-edb789b313e3/20220227_103929/sct-runner-4cc6ea94.tar.gz)&source=gmail-html&ust=1646143203458000&usg=AOvVaw1fkSAGxDmvN4NXtWYlvQVP)


[Jenkins job URL]([https://jenkins.scylladb.com/job/scylla-master/job/longevity/job/longevity-100gb-4h-ebs-gp3-test/36/](https://www.google.com/url?q=https://jenkins.scylladb.com/job/scylla-master/job/longevity/job/longevity-100gb-4h-ebs-gp3-test/36/&source=gmail-html&ust=1646143203458000&usg=AOvVaw2M6w08SyYpkhUERSb8BEnX))

**Comment by kbr-** [comment]
It looks like the decommissioned node's tokens were not removed from the token ring for some reason, so other nodes still considered it as a replica.

Is it just a race? Maybe it takes some time for other nodes to clean up after the decommissioned node?
@asias what do you think?

**Comment by asias** [comment]
@kbr-

The new node already learned that the node 10.0.1.235 has been removed from the cluster.  You can check why CDC still insists to write to 6 nodes.  

On the new node, the following means it has learned the status for 10.0.1.235 is LEFT. 
```
2022-02-20T05:36:47+00:00 db-node-490f8141-7 !    INFO |  [shard 0] gossip - Node 10.0.1.235 will be removed from gossip at [2022-02-23 05:30:30]: (expire = 1645594230175499179, now = 1645335407766195375,
```

**Comment by kbr-** [comment]
CDC simply performs a regular CL=ALL write to a keyspace which uses `EverywhereReplicationStrategy`. The strategy calculates replicas using `token_metadata::get_all_endpoints`:
```cpp
inet_address_vector_replica_set everywhere_replication_strategy::get_natural_endpoints(const token&, const effective_replication_map& erm) const {
    const auto& tm = *erm.get_token_metadata_ptr();
    if (tm.sorted_tokens().empty()) {
        return inet_address_vector_replica_set({utils::fb_utilities::get_broadcast_address()});
    }
    return boost::copy_range<inet_address_vector_replica_set>(tm.get_all_endpoints());
}
```

But isn't 6 replicas actually correct? The 5 normal ones + the pending one (the joining node). Indeed, pending ranges are updated before the CDC generation is created:
```cpp
      if (!bootstrap_rbno) {
        // When is_repair_based_node_ops_enabled is true, the bootstrap node
        // will use node_ops_cmd to bootstrap, node_ops_cmd will update the pending ranges.
        slogger.debug("bootstrap: update pending ranges: endpoint={} bootstrap_tokens={}", get_broadcast_address(), _bootstrap_tokens);
        mutate_token_metadata([this] (mutable_token_metadata_ptr tmptr) {
            auto endpoint = get_broadcast_address();
            tmptr->add_bootstrap_tokens(_bootstrap_tokens, endpoint);
            return update_pending_ranges(std::move(tmptr), format("bootstrapping node {}", endpoint));
        }).get();
      }

        // After we pick a generation timestamp, we start gossiping it, and we stick with it.
        // We don't do any other generation switches (unless we crash before complecting bootstrap).
        assert(!_cdc_gen_id);

        _cdc_gen_id = _cdc_gen_service.local().make_new_generation(_bootstrap_tokens, !is_first_node()).get0();
```

In that case I have a simpler explanation - there was a transient network problem that caused one of the replicas to not respond.

It's sad though that the error message doesn't tell us which replicas failed to respond.

**Comment by kbr-** [comment]
I confirmed that 6 responses is the correct and expected number for this write: there are 5 normal nodes and 1 joining node, and the write exception reports the number of all replicas, including pending replicas.

My hypothesis is that one replicas had problems (not necessarily CDC related, just general) and simply did not manage to perform the write, hence we got a timeout.

I checked the metrics and indeed the explanation is plausible, there is a replica that had problems and was timing out writes all the time throughout the test, 10.0.2.60 (`longevity-100gb-4h-master-db-node-490f8141-4`):
![image](https://user-images.githubusercontent.com/6516501/158615573-e8507803-c6c6-4ea4-a9d5-3aad758fc9cc.png)

it seems our CDC generation write got unlucky and joined these failed writes (the write started at `05:37:36` and timed out a minute later).

---

In general if the CDC generation write during bootstrap fails, it will abort the bootstrap. The administrator should then:
1. ensure network connectivity between cluster
2. try to get nodes to a healthy state
3. retry bootstrap.

I'm closing the issue because there's nothing unexpected with CDC going on here. If someone wants to investigate further e.g. why 10.0.2.60 was misbehaving during the test, they are free to reopen it.
